src/main/java/com/google/edith/DealItem.java

```diff
@@ -91,6 +95,44 @@ public void setComment(String comment) {
     this.comment = comment;
   }
 
+  /**
+   * Input string expiration is expected to have the minimum expiration time, the maximum expiration
+   * time, and the unit of time, in no particular order. It parses the string to find the minimum
+   * expiration time and the unit, as that is the data needed to generate new grocery lists based
+   * off of.
+   *
+   * <p>ex: '1.0 2.0 Weeks' -> this.expiration = '1.0 Weeks'
+   */
+  public void setExpiration(String expiration) {
+    if (expiration.equals("NO_EXPIRATION")) {
+      this.expiration = expiration;
+      return;
+    }
+
+    String[] expirationPieces = expiration.split(" ");
+    List<Double> range = new ArrayList<Double>();
+    String timeMeasurement = "";
+
+    for (String expirationPiece : expirationPieces) {
+      try {
+        range.add(Double.parseDouble(expirationPiece));
+      } catch (NumberFormatException e) {
+        timeMeasurement = expirationPiece;
+      }
+    }
+
+    Double min = new Double(0);
+    if (range.size() >= 1) {
+      min = range.get(0);
+    }
+    for (Double time : range) {
+      if (time < min) {
+        min = time;
+      }
+    }
+    this.expiration = min + " " + timeMeasurement;
```

Comment:
This might be cleaner use a static string that gets formatted.

```
private static final EXPIRATION_TEMPLATE = "%s %s";
this.expiration = String.format(EXPIRATION_FORMAT, min, timeMeasurement);